data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilter.java

```diff
@@ -297,7 +297,9 @@ public void testNoNulls() {
   }
 
   @Test
-  public void testIsNaN() {
+  public void testIsNaNForParquet() {
```

Comment:
Nit: You don't need to change the test name, because the `Assume` will take care of it and signal that in the tests.


For people that maintain forks and cherry-pick individual changes, updating the test name will probably cause them more of headache than is worth it.

Also, I would consider using `Assume.assumeTrue("Parquet metrics do not track NaN", format == FileFormat.PARQUET)` unless this should also possibly run someday for Avro.